src/language-js/printer-estree.js

```diff
@@ -553,11 +554,13 @@ function printPathNoParens(path, options, print, args) {
       );
     case "BinaryExpression":
     case "LogicalExpression":
-    case "NGPipeExpression": {
+    case "NGPipeExpression":
+    case "TSAsExpression": {
+      const node = convertToBinaryishNode(n);
       const parent = path.getParentNode();
       const parentParent = path.getParentNode(1);
       const isInsideParenthesis =
-        n !== parent.body &&
+        node !== parent.body &&
```

Comment:
This condition becomes broken. You compare the original `n` with `parent.*`.

Reply:
But let's better eliminate `convertToBinaryishNode` and use variables like `leftNodeName` and `rightNodeName` instead. The code would be cleaner that way IMHO. For example, such mistakes (`n` vs `node`) wouldn't be possible.